modules/ROOT/pages/design-common-errors-08.adoc

```diff
@@ -0,0 +1,360 @@
+= Common Errors in Conforming to RAML 0.8
+
+When testing its new RAML parser (which was released on October 24, 2018), MuleSoft identified ten common errors in API specifications written in RAML 0.8. This topic explains the errors with examples and describes how to fix them. They are not listed in any particular order, such as frequency of occurrence in the sample of API specifications that MuleSoft tested.
+
+If you open in the code editor a published API specification that is written in RAML 0.8 and you see error indicators, this topic might help you to resolve most of the errors. If you see errors that this topic does not mention, refer to the specification for RAML 0.8, which you can read on https://raml.org/[raml.org].
+
+== Common Error 1
+// APIMF-824
+
+*_Not using a data type of string for the example of a Boolean enum_*
+
+If the values for a Boolean enum are either both strings or both Boolean values, the example of the enum must be a string. To illustrate the error, this snippet shows an example using the wrong data type.
+
+----
+#%RAML 0.8
+title: ExampleRAML
+/nob:
+  head:
+    responses:
+      204:
+        headers:
+          X-NOB-Exists:
+            enum: [true, false]
+            example: true
+----
+
+
+The data type for this example is correct:
+
+
+----
+...
+            enum: [true, false]
+            example: "true"
+----
+
+The data type for this example is also correct:
+
+
+----
+...
+            enum: ["true", "false"]
+            example: "true"
+----
+
+== Common Error 2
+// APIMF-841
+
+*_Including a schema that contains invalid JSON_*
+
+The JSON in files that are included in the value of the `schemas` property must be valid.
+
+The first snippet includes the schema `appSwitcher.json`.  However, the second shows that there is an error in the JSON: an extra comma after the last key/value pair. The editor flags the error in the specification on the line with the `!include` statement.
+
+
+----
+#%RAML 0.8
+title: ExampleRAML
+schemas:
+  - appSwitcher: !include schemas/appSwitcher.json
+----
+
+
+----
+{
+  "appMenuItems" : [
+    {
+      "type" : "Tabset" ,
+      "content" : null ,
+      "icons" : null ,
+      "colors" : null ,
+      "label" : "Call Center" ,
+      "url" : "/home/home.jsp?tsid=02uxx00000056Sr"
+    } , {
+      "type" : "Tabset" ,
+      "content" : null ,
+      "icons" : null ,
+      "colors" : null ,
+      "label" : "Community" ,
+      "url" : "/home/home.jsp?tsid=02uxx00000056Sw"
+    } , {
+      "type" : "Tabset" ,
+      "content" : null ,
+      "icons" : null ,
+      "colors" : null ,
+      "label" : "App Launcher" ,
+      "url" : "/app/mgmt/applauncher/appLauncher.apexp?tsid=02uxx00000056Sx",
+    }
+  ]
+}
+----
```

Comment:
We support the last comma in a map, although if the json file was invalid the error should appear under the location appSwitcher.json and not in the include tag of the main raml.

Reply:
I've changed this sentence: "However, the second shows that there is an error in the JSON: an extra comma after the last key/value pair." It now reads "However, the second shows that there is an error in the JSON: a quotation mark is missing at the end of the last value." And I removed the quotation mark.
I've also removed the sentence "The editor flags the error in the specification on the line with the `!include` statement."
You wrote: "if the json file was invalid the error should appear under the location appSwitcher.json". I don't understand where this is.

Reply:
Each validation result contains a location that specify the file where the error has been detected. I was trying to say that in this case the location for any error in the json will have has location the appSwitecher.json file.

Reply:
As far as I know, we aren't exposing validation results to users, but only individual error and warning messages. Therefore, I won't need to document what validation results include about the locations of errors.